lib/common/zstd_internal.h

```diff
@@ -219,6 +219,12 @@ typedef struct rawSeq_s {
     U32 matchLength;
 } rawSeq;
 
+typedef struct {
+  rawSeq* seq;
+  size_t nbSeq;
+  size_t pos;
+} rawSeqStore_t;
```

Comment:
It seems `rawSeqStore_t` is only needed within `/lib/compress` (which feels logical).
In which case, it would be more accurate to define it within `/lib/compress/zstd_compress_internal.h`.

Reply:
I regularly had hesitations about the meaning of `nbSeq`.
Sometimes, it's the capacity of `seq`, sometimes it's the nb of sequences actually present within `seq`, depending on read / write mode.

This might be because the internal API is regularly switching between `rawSeq*` and `rawSeqStore_t*` types, with `seqStore` becoming a relatively short-lived container.

I believe clarity would be helped by separating concepts `seqCapacity` and `nbSeq`.

Reply:
I'm changing `nbSeq` to `seqCapacity` where appropriate, and `rawSeqStore_t::nbSeq` to `rawSeqStore_t::end`.

The only function that doesn't take the `rawSeqStore_t` is `ZSTD_ldm_generateSequences()`, since it can use the return value, and doesn't support streaming. Do you think it would help clarity to take `rawSeqStore_t*` and store the number of generated sequences in `rawSeqStore_t::pos` instead?

Reply:
Yes, I believe so.

In isolation, the current design of `ZSTD_ldm_generateSequences()` would be fine,
but within a family of functions using `rawSeqStore_t*` as argument,
it seems better to keep the same type within interface.